modules/vba_documents/spec/mailers/unsuccessful_report_mailer_spec.rb

```diff
@@ -33,31 +33,59 @@
   end
 
   describe '#build' do
-    subject do
-      described_class.build(totals, [error_upload], [uploaded_upload], 7.days.ago, Time.zone.now).deliver_now
+    module MailerHelper
+      class << self
+        attr_reader :env_hash
+      end
+
+      @env_hash = {
+        staging: 'https://dsva-vetsgov-staging-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        dev: 'https://dsva-vetsgov-dev-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        sandbox: 'https://dsva-vagov-sandbox-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        prod: 'https://dsva-vetsgov-prod-benefits-documents.s3.us-gov-west-1.amazonaws.com/'
+      }
     end
 
-    it 'sends the email' do
-      expect(subject.subject).to eq('Benefits Intake Unsuccessful Submission Report')
+    before do
+      current_env_url = env_url.last
+      Settings.vba_documents.location.prefix = current_env_url
+      VBADocuments::Deployment.environment = VBADocuments::Deployment.fetch_environment
+      if VBADocuments::UnsuccessfulReportMailer.const_defined?(:RECIPIENTS)
+        VBADocuments::UnsuccessfulReportMailer.send(:remove_const, :RECIPIENTS)
+      end
+      load './modules/vba_documents/app/mailers/vba_documents/unsuccessful_report_mailer.rb'
+      @email = described_class.build(totals, [error_upload], [uploaded_upload], 7.days.ago, Time.zone.now).deliver_now
```

Comment:
I'm unsure of this delivering the emails within the test. It could uncover e2e errors. If this is an e2e test then I'm on board.

Reply:
What is e2e?
It is not clear to me that your worries were not present before these changes.  In other words, there was already an email test, as opposed to one per environment.